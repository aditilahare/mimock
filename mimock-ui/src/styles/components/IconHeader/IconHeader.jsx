import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@mui/material/Tooltip';
import { useNavigate } from 'react-router-dom';
import { IoMdArrowRoundBack } from 'react-icons/io';
import { IconHeaderWrapper, Title, GoBack } from './IconHeader.style';

function IconHeader({
	icon,
	dataTestId,
	title,
	enableBackNavigation,
	additionalStyles,
}) {
	const navigate = useNavigate();

	return (
		<IconHeaderWrapper
			data-testid={dataTestId}
			$additionalStyles={additionalStyles}
		>
			<If condition={enableBackNavigation}>
				<Tooltip title='Go back' arrow>
					<GoBack
						data-testid='go-back-btn'
						onClick={() => {
							navigate(-1);
						}}
					>
						<IoMdArrowRoundBack />
					</GoBack>
				</Tooltip>
			</If>
			{icon}
			<Title>{title}</Title>
		</IconHeaderWrapper>
	);
}

IconHeader.propTypes = {
	icon: PropTypes.element.isRequired,
	dataTestId: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	enableBackNavigation: PropTypes.bool,
	additionalStyles: PropTypes.string,
};

IconHeader.defaultProps = {
	enableBackNavigation: false,
};

export default IconHeader;
