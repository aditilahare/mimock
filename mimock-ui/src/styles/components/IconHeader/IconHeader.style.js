import tw from 'tailwind-styled-components';
import IconButton from '@mui/material/IconButton';
import styled from 'styled-components';

export const IconHeaderWrapper = tw.div`
  flex
  items-center
  align-middle
  font-sans
  text-2xl
  font-bold
  text-gray-700
`;

export const Title = tw.div`
  mx-2
  select-none
`;

export const GoBack = styled(IconButton)`
	margin-left: 10px !important;
	margin-right: 10px !important;
`;
