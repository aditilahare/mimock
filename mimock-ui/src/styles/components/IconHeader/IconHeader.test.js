import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { FaHome } from 'react-icons/fa';
import IconHeader from './IconHeader';

let mockUseNavigation = jest.fn();
jest.mock('react-router-dom', () => ({
	useNavigate: () => mockUseNavigation,
}));

describe('IconHeader', () => {
	it('should render icon header component without back button', async () => {
		const { getByTestId, container, queryByTestId } = await render(
			<IconHeader icon={<FaHome />} title='Home' dataTestId='home-header' />
		);

		expect(getByTestId('home-header')).toBeInTheDocument();
		expect(getByTestId('home-header')).toHaveTextContent('Home');
		expect(queryByTestId('go-back-btn')).toBeNull();

		expect(container).toMatchSnapshot();
	});

	it('should render icon header component with back button', async () => {
		const { getByTestId, container } = await render(
			<IconHeader
				icon={<FaHome />}
				title='Home'
				dataTestId='home-header'
				enableBackNavigation={true}
			/>
		);

		expect(getByTestId('home-header')).toBeInTheDocument();
		expect(getByTestId('home-header')).toHaveTextContent('Home');

		await act(async () => {
			fireEvent.click(getByTestId('go-back-btn'));
		});

		expect(mockUseNavigation).toHaveBeenCalledTimes(1);
		expect(mockUseNavigation).toHaveBeenCalledWith(-1);

		expect(container).toMatchSnapshot();
	});
});
