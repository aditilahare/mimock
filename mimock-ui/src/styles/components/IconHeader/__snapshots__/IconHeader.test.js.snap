// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconHeader should render icon header component with back button 1`] = `
<div>
  <div
    class="flex items-center align-middle font-sans text-2xl font-bold text-gray-700"
    data-testid="home-header"
  >
    <button
      aria-label="Go back"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-bdvvtL glRTRM css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      data-testid="go-back-btn"
      tabindex="0"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M401.4 224h-214l83-79.4c11.9-12.5 11.9-32.7 0-45.2s-31.2-12.5-43.2 0L89 233.4c-6 5.8-9 13.7-9 22.4v.4c0 8.7 3 16.6 9 22.4l138.1 134c12 12.5 31.3 12.5 43.2 0 11.9-12.5 11.9-32.7 0-45.2l-83-79.4h214c16.9 0 30.6-14.3 30.6-32 .1-18-13.6-32-30.5-32z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
      />
    </svg>
    <div
      class="mx-2 select-none"
    >
      Home
    </div>
  </div>
</div>
`;

exports[`IconHeader should render icon header component without back button 1`] = `
<div>
  <div
    class="flex items-center align-middle font-sans text-2xl font-bold text-gray-700"
    data-testid="home-header"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
      />
    </svg>
    <div
      class="mx-2 select-none"
    >
      Home
    </div>
  </div>
</div>
`;
